wp-content/mu-plugins/pub/wporg-learn-cli.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+if ( ! defined( 'WP_CLI' ) ) {
+	return;
+}
+
+/**
+ * Converts a tutorial post to a lesson post type.
+ */
+class WPORG_Learn_Tutorial_To_Lesson_Command extends WP_CLI_Command {
+	/**
+	 * Converts tutorials to lessons based on the provided URL or file.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <source>
+	 * : The URL of the tutorial post to convert, or path to a file containing URLs (one per line)
+	 *
+	 * [--live]
+	 * : Actually perform the conversion (default is dry-run)
+	 *
+	 * [--file]
+	 * : Indicates that the source is a file containing URLs
+	 *
+	 * ## EXAMPLES
+	 *
+	 * wp wporg-learn-tutorial-to-lesson convert https://learn.wordpress.org/tutorial/slug
+	 * wp wporg-learn-tutorial-to-lesson convert urls.txt --file
+	 * wp wporg-learn-tutorial-to-lesson convert urls.txt --file --live
+	 */
+	public function convert( $args, $assoc_args ) {
+		$source = $args[0];
+		$is_dry_run = ! isset( $assoc_args['live'] );
+		$is_file = isset( $assoc_args['file'] );
+
+		$urls = array();
+		if ( $is_file ) {
+			if ( ! file_exists( $source ) ) {
+				WP_CLI::error( "File not found: $source" );
+				return;
+			}
+			$urls = array_filter( explode( "\n", file_get_contents( $source ) ) );
+		} else {
+			$urls = array( $source );
+		}
+
+		foreach ( $urls as $url ) {
```

Comment:
some consideration should be made for memory expensive operations when looping through an indeterminate number of items